src/Confluent.SchemaRegistry.Serdes.Json/JsonSerializer.cs

```diff
@@ -246,7 +216,28 @@ public async Task<byte[]> SerializeAsync(T value, SerializationContext context)
                 }
                 finally
                 {
-                    serializeMutex.Release();
+                    serdeMutex.Release();
```

Comment:
I believe one reason is because the serializer is used by the producer, which is multithreaded, whereas the deserializer is used by the consumer, which is not multithreaded